request_wrapper.py

```diff
@@ -0,0 +1,68 @@
+
+import subprocess
+import pycurl
+import json
+from StringIO import StringIO 
+import os
+
+class RequestWrapper:
+    """ Wrapper for making http requests to xsdb api """
+
+    base_url = 'https://cms-gen-dev.cern.ch/xsdb'
+    api_url = base_url + '/api'
+
+    subprocess.call(['bash', 'getCookie.sh'])
+
+    c = pycurl.Curl()
+    c.setopt(pycurl.FOLLOWLOCATION, 1)
+    c.setopt(pycurl.COOKIEJAR, os.path.expanduser("~/private/xsdbdev-cookie.txt"))
+    c.setopt(pycurl.COOKIEFILE, os.path.expanduser("~/private/xsdbdev-cookie.txt"))
+    c.setopt(pycurl.HTTPHEADER, ['Content-Type:application/json', 'Accept:application/json'])
+    #c.setopt(pycurl.VERBOSE, 0)
+    def simple_search_to_dict(self, keyval_dict):
+        rslt = self._perform_post(self.api_url + '/search', keyval_dict)
+        return rslt
+
+    def adv_search(self, keyval_dict={}, page_size=20, current_page=0, orderby_field="", order_direction=1):
```

Comment:
Line is too long, you can split it in two  or more lines (just don't leave trailing whitespaces).
PEP8 on how to split long lines: https://www.python.org/dev/peps/pep-0008/#indentation
```
C: 26, 0: Line too long (108/100) (line-too-long)
```

Reply:
```
C: 26, 4: Missing method docstring (missing-docstring)
```

Reply:
It is actually a bad idea to use `{}` (or list, or set or any other mutable object) as a default parameter. Because this way, this dictionary is shared among all calls, that is, if you add something to this dictionary during function call, that item will still be present in this dictionary during all subsequent calls. I see that you don't modify it in this method, however, it is considered dangerous and it is advised to avoid.
Correct way is to do `keyval_dict=None` in method definition and then add this in method body:
```
if keyval_dict is None:
    keyval_dict = {}
```
Few examples: https://stackoverflow.com/questions/26320899/
```
W: 26, 4: Dangerous default value {} as argument (dangerous-default-value)
```